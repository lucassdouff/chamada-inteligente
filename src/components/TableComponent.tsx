import React from "react";
import { ScrollView, View, Text } from "react-native";
import { TouchableWithoutFeedback } from "react-native-gesture-handler";
import { TableDataModel } from "../core/models/TableDataModel";

interface TableComponentProps {
    tableData: TableDataModel[][];
}

export default function TableComponent({ tableData } : TableComponentProps) {

    return (
        <View className="w-full">
            <ScrollView horizontal={true}>
                <View className="flex-col border divide-y"> 
                    {
                        tableData.map((arrList) => (
                            <View className="flex-row divide-x">
                                {
                                    arrList.map((item) => (
                                        <View>
                                            {
                                                item.action ? 
                                                <TouchableWithoutFeedback onPress={item.action}>
                                                    <Text className="w-24 py-2 px-2.5 text-center underline">{item.text}</Text>
                                                </TouchableWithoutFeedback>
                                                : <Text className="w-24 py-2 px-2.5 text-center">{item.text}</Text>
                                            }
                                        </View>
                                    ))
                                }
                            </View>
                        ))
                    }
                </View>
            </ScrollView>
        </View>
    )
}