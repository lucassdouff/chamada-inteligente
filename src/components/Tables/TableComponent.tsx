import React from "react";
import { ScrollView, View, Text, Button } from "react-native";
import { TableDataModel } from "../../core/models/TableDataModel";

interface TableComponentProps {
    tableData: TableDataModel[][] | undefined;
}

export default function TableComponent({ tableData } : TableComponentProps) {

    const tableDataWithoutRemoveLine = tableData?.map((arrList) => (
        arrList.filter((item) => !item.removeLine)
    ));

    return (
        <View className="w-full">
            <ScrollView horizontal={true}>
                <View className="flex-col border divide-y"> 
                    {
                        tableDataWithoutRemoveLine?.map((arrList,index) => (
                            <View key={index} className="flex-row divide-x">
                                {
                                    arrList.map((item,index) => (
                                        <View key={index} className="flex items-center justify-center">
                                            {
                                                item.action ? 
                                                <View className="m-4 py-1.5 px-2">
                                                    <Button testID="table-action" title={item.text} color='blue' onPress={item.action} />
                                                </View>
                                                : <Text className="m-2 w-24 py-1.5 px-2 text-center">{item.text}</Text>
                                            }
                                        </View>
                                    ))
                                }
                            </View>
                        ))
                    }
                </View>
            </ScrollView>
        </View>
    )
}