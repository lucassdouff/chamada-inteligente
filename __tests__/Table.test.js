import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';
import TableComponent from '../src/components/Tables/TableComponent';

describe('Table component should render correctly and do some action', () => {
    it('renders correctly', () => {
        const table = renderer.create(<TableComponent 
            tableData={[
                [
                    {
                        removeLine: true,
                        text: 'Student 1',
                        action: () => {},
                    }
                ],
                [
                    {
                        removeLine: false,
                        text: 'Student 2',
                        action: () => {},
                    }
                ],
                [
                    {
                        removeLine: false,
                        text: 'Student 3',
                        action: () => {},
                    }
                ],
            ]}
        />).toJSON();
        
        expect(table).toMatchSnapshot();
    });

    it('performs action on click', () => {
        const mockAction = jest.fn();
        const { getByTestId } = render(<TableComponent 
            tableData={[
                [
                    {
                        removeLine: true,
                        text: 'Student 1',
                    }
                ],
                [
                    {
                        removeLine: false,
                        text: 'Student 2',
                    }
                ],
                [
                    {
                        removeLine: false,
                        text: 'Student 3',
                        action: mockAction,
                    }
                ],
            ]}
        />);
        fireEvent.press(getByTestId('table-action'));
        expect(mockAction).toHaveBeenCalled();
    });
});
