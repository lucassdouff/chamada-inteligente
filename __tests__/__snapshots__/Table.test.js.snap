// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component should render correctly and do some action renders correctly 1`] = `
<View
  style={
    [
      {
        "width": "100%",
      },
    ]
  }
>
  <RCTScrollView
    horizontal={true}
  >
    <View>
      <View
        style={
          [
            {
              "flexDirection": "column",
            },
            {
              "borderBottomWidth": 1,
              "borderLeftWidth": 1,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
            ]
          }
        />
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              [
                {
                  "borderBottomWidth": 0,
                  "borderTopWidth": 1,
                },
              ],
            ]
          }
        >
          <View
            style={
              [
                {
                  "display": "flex",
                },
                {
                  "alignItems": "center",
                },
                {
                  "justifyContent": "center",
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "marginBottom": 16,
                    "marginLeft": 16,
                    "marginRight": 16,
                    "marginTop": 16,
                  },
                  {
                    "paddingBottom": 6,
                    "paddingTop": 6,
                  },
                  {
                    "paddingLeft": 8,
                    "paddingRight": 8,
                  },
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="table-action"
              >
                <View
                  style={
                    [
                      {},
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "margin": 8,
                          "textAlign": "center",
                        },
                        {
                          "color": "blue",
                        },
                      ]
                    }
                  >
                    Student 2
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              [
                {
                  "borderBottomWidth": 0,
                  "borderTopWidth": 1,
                },
              ],
            ]
          }
        >
          <View
            style={
              [
                {
                  "display": "flex",
                },
                {
                  "alignItems": "center",
                },
                {
                  "justifyContent": "center",
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "marginBottom": 16,
                    "marginLeft": 16,
                    "marginRight": 16,
                    "marginTop": 16,
                  },
                  {
                    "paddingBottom": 6,
                    "paddingTop": 6,
                  },
                  {
                    "paddingLeft": 8,
                    "paddingRight": 8,
                  },
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="table-action"
              >
                <View
                  style={
                    [
                      {},
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "margin": 8,
                          "textAlign": "center",
                        },
                        {
                          "color": "blue",
                        },
                      ]
                    }
                  >
                    Student 3
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
